src/commands/force/mdapi/beta/deploy.ts

```diff
@@ -0,0 +1,221 @@
+      exactlyOne: ['validateddeployrequestid', 'deploydir'],
+    }),
+    singlepackage: flags.boolean({
+      char: 's',
+      description: messages.getMessage('flags.singlePackage'),
+      longDescription: messages.getMessage('flagsLong.singlePackage'),
+    }),
+    soapdeploy: flags.boolean({
+      description: messages.getMessage('flags.soapDeploy'),
+      longDescription: messages.getMessage('flagsLong.soapDeploy'),
+    }),
+  };
+  private isAsync = false;
+  private isRest = false;
+  private asyncDeployResult: AsyncResult;
+
+  private updateDeployId = once((id: string) => {
+    this.displayDeployId(id);
+    this.setStash(id);
+  });
+
+  public async run(): Promise<MdDeployResult | DeployCommandAsyncResult> {
+    // start deploy with zip if not already an ID
+    // report on that ID
+    await this.deploy();
+    this.resolveSuccess();
+    return this.formatResult();
+  }
+
+  protected async deploy(): Promise<void> {
+    this.isSourceStash = false;
+    const waitDuration = this.getFlag<Duration>('wait');
+    this.isAsync = waitDuration.quantity === 0;
+    this.isRest = await this.isRestDeploy();
+
+    if (this.flags.validateddeployrequestid) {
+      this.deployResult = await this.deployRecentValidation();
+      return;
+    }
+    const deploymentOptions = this.flags.zipfile
+      ? { zipPath: this.flags.zipfile as string }
+      : { mdapiPath: this.flags.deploydir as string };
+
+    // still here?  we need to deploy a zip file then
+    const deploy = new MetadataApiDeploy({
+      usernameOrConnection: this.org.getUsername(),
+      ...deploymentOptions,
+      apiOptions: {
+        ignoreWarnings: this.getFlag('ignorewarnings', false),
+        rollbackOnError: !this.getFlag('ignoreerrors', false),
+        checkOnly: this.getFlag('checkonly', false),
+        runTests: this.getFlag<string[]>('runtests'),
+        testLevel: this.getFlag<TestLevel>('testlevel'),
+        singlePackage: this.getFlag('singlepackage', false),
+        rest: this.isRest,
+      },
+    });
+    await deploy.start();
+    this.asyncDeployResult = { id: deploy.id };
+    this.updateDeployId(deploy.id);
+
+    // we might not know the source api version without unzipping a zip file, so we don't use componentSet
+    this.ux.log(getVersionMessage('Deploying', undefined, this.isRest));
+
+    this.logger.debug('Deploy result: %o', deploy);
+    this.updateDeployId(deploy.id);
+
+    if (!this.isAsync) {
+      if (!this.isJsonOutput()) {
+        const progressFormatter: ProgressFormatter = env.getBoolean('SFDX_USE_PROGRESS_BAR', true)
+          ? new DeployProgressBarFormatter(this.logger, this.ux)
+          : new DeployProgressStatusFormatter(this.logger, this.ux);
+        progressFormatter.progress(deploy);
+      }
+      this.displayDeployId(deploy.id);
+      this.deployResult = await deploy.pollStatus(500, waitDuration.seconds);
+      // this.deployResult = await this.report(this.asyncDeployResult.id);
```

Comment:
why aren't we setting `this.deployResult`?

Reply:
it is...I just left the previous version in.  `deploy` now uses `pollStatus`